benchmarks/cugraph/pytest-based/bench_algos.py

```diff
@@ -43,41 +47,104 @@ def setFixtureParamNames(*args, **kwargs):
+                                      store_transposed=store_transposed)
+        else:
+            G.from_cudf_edgelist(df,
+                                 source="src",
+                                 destination="dst",
+                                 edge_attr=edge_attr,
+                                 store_transposed=store_transposed)
+        return G
+
+    def get_path(self):
+        """
+        (this is likely not needed for use with pytest-benchmark, just added for
+        API completeness with Dataset.)
+        """
+        return str(self)
+
+    def unload(self):
+        self._edgelist = None
+
+
+_rmat_scale = getattr(pytest, "_rmat_scale", 20)  # ~1M vertices
+_rmat_edgefactor = getattr(pytest, "_rmat_edgefactor", 16)  # ~17M edges
+rmat_sg_dataset = pytest.param(RmatDataset(scale=_rmat_scale,
+                                           edgefactor=_rmat_edgefactor,
+                                           mg=False),
+                               marks=[pytest.mark.rmat_data,
+                                      pytest.mark.sg,
+                               ])
+rmat_mg_dataset = pytest.param(RmatDataset(scale=_rmat_scale,
+                                           edgefactor=_rmat_edgefactor,
+                                           mg=True),
+                               marks=[pytest.mark.rmat_data,
+                                      pytest.mark.mg,
+                               ])
+
+rmm_fixture_params = gen_fixture_params_product(
     (managed_memory, "mm"),
     (pool_allocator, "pa"))
-
-###############################################################################
-# Helpers
-def createGraph(csvFileName, graphType=None):
-    """
-    Helper function to create a Graph (directed or undirected) based on
-    csvFileName.
-    """
-    if graphType is None:
-        # There's potential value in verifying that a directed graph can be
-        # created from a undirected dataset, and an undirected from a directed
-        # dataset. (For now?) do not include those combinations to keep
-        # benchmark runtime and complexity lower, and assume tests have
-        # coverage to verify correctness for those combinations.
-        if "directed" in csvFileName.parts:
-            graphType = cugraph.Graph(directed=True)
-        else:
-            graphType = cugraph.Graph()
-
-    gdf = utils.read_csv_file(csvFileName)
-    if len(gdf.columns) == 2:
-        edge_attr = None
-    else:
-        edge_attr = "2"
-
-    return cugraph.from_cudf_edgelist(
-        gdf,
-        source="0", destination="1", edge_attr=edge_attr,
-        create_using=graphType,
-        renumber=True)
-
+dataset_fixture_params = gen_fixture_params_product(
+    (directed_datasets +
+     undirected_datasets +
+     [rmat_sg_dataset, rmat_mg_dataset], "ds"))
 
 # Record the current RMM settings so reinitialize() will be called only when a
 # change is needed (RMM defaults both values to False). The --allow-rmm-reinit
```

Comment:
Is this where we should be setting the `cupy` and `torch` allocators as well?

Reply:
`cupy` is set to use rmm pool when we import `cuDF`. 

 We can decide to add `pytorch` based on wether we decide to run `dgl/pyg ` benchmarks here.

Reply:
Two questions:

1. Is this config for SG or for MG too ? 
2. If only SG, it might be worth exploring setting `pool_alloc=True` to speed up CI (if that is a concern).

Reply:
Thanks for looking into this, here's some background:

`bench_algos.py` was originally SG-only, and this PR adds MG coverage. The parameterization of managed mem and pool alloc adds (potentially a lot of) benchmark run time and we've been leaving the defaults in place for automated runs. The use case of actually changing the RMM config multiple times in the same process also did not seem to match a typical user's workflow and sometimes led to problems if we didn't clean up properly (as you pointed out elsewhere), so having pytest run through all combinations is only done using a special CLI option `--allow-rmm-reinit` (see [here](https://github.com/rapidsai/cugraph/blob/f4433f499403aa04e0393e9cef21306cbb5497ea/benchmarks/cugraph/pytest-based/conftest.py#L18)). Otherwise, if a user wants to change the default they just pick the marker they want to use.  The current defaults are pool_alloc `True`, managed_mem `False` (see [here](https://github.com/rapidsai/cugraph/blob/f4433f499403aa04e0393e9cef21306cbb5497ea/benchmarks/cugraph/pytest-based/conftest.py#L58)).  

To answer @VibhuJawa 's questions more directly:
1. it's for SG, I should add a FIXME to figure out how to apply these settings to the dask config when using MG. This is actually a big gap IMO which I'll address either in this PR or a follow-up.
2. pool_alloc defaults to True in conftest.py, but I should add a comment here mentioning that.

To answer @alexbarghi-nv 's question:
> Is this where we should be setting the cupy and torch allocators as well?

I like @VibhuJawa 's suggestion ("We can decide to add pytorch based on wether we decide to run dgl/pyg benchmarks here.")

Reply:
Thanks for the detailed answer. It makes sense to me.